pkg/types/ccipocr3/plugin_commit_types.go

```diff
@@ -30,21 +33,32 @@ func (r CommitPluginReport) IsEmpty() bool {
 		len(r.RMNSignatures) == 0
 }
 
+// MerkleRootChain Mirroring https://github.com/smartcontractkit/chainlink/blob/cd5c78959575f593b27fd83d8766086d0c678487/contracts/src/v0.8/ccip/libraries/Internal.sol#L356-L362
 type MerkleRootChain struct {
-	ChainSel     ChainSelector `json:"chain"`
-	SeqNumsRange SeqNumRange   `json:"seqNumsRange"`
-	MerkleRoot   Bytes32       `json:"merkleRoot"`
+	ChainSel      ChainSelector `json:"chain"`
+	OnRampAddress Bytes         `json:"onRampAddress"`
+	SeqNumsRange  SeqNumRange   `json:"seqNumsRange"`
+	MerkleRoot    Bytes32       `json:"merkleRoot"`
+}
+
+func (m MerkleRootChain) Equals(other MerkleRootChain) bool {
+	return m.ChainSel == other.ChainSel &&
+		bytes.Equal(m.OnRampAddress, other.OnRampAddress) &&
+		m.SeqNumsRange == other.SeqNumsRange &&
+		m.MerkleRoot == other.MerkleRoot
```

Comment:
```suggestion
		bytes.Equal(m.MerkleRoot[:], other.MerkleRoot[:])
```

Reply:
Is there a semantic difference b/w your suggestion and the current code?

Reply:
No, I guess arrays can use `==`